app/javascript/angular/code/services/_mobile_sessions.js

```diff
@@ -88,60 +90,70 @@ export const mobileSessions = (
     },
 
     deselectSession: function() {
-      const session = sessionsUtils.selectedSession(this);
-      if (!session) return;
-      session.loaded = false;
+      if (!sessionsUtils.isSessionSelected()) return;
+      this.selectedSession = {};
       params.update({ prevMapPosition: {} });
       params.update({ selectedSessionIds: [] });
-      drawSession.undoDraw(session, prevMapPosition);
+      drawSession.undoDraw({}, prevMapPosition);
     },
 
     selectSession: function(id) {
-      const callback = session => data => {
-        drawSession.clear(this.sessions);
+      const callback = rawData => {
+        this.selectedSession = prepareSessionData(rawData);
+        drawSession.clear(this.sessions); //todo probable doesnt work
         prevMapPosition = {
           bounds: map.getBounds(),
           zoom: map.getZoom()
         };
         params.update({ prevMapPosition: prevMapPosition });
 
-        const drawSessionStartingMarker = (session, sensorName) =>
-          this.drawSessionWithLabel(session, sensorName);
-        const draw = () =>
-          drawSession.drawMobileSession(session, drawSessionStartingMarker);
         map.fitBoundsWithBottomPadding(
-          calculateBounds(sensors, session, map.getZoom())
+          calculateBounds(sensors, this.selectedSession, map.getZoom())
+        );
+
+        const drawSessionStartingMarker = (selectedSession, sensorName) =>
+          this.drawSessionWithLabel(selectedSession, sensorName);
+        drawSession.drawMobileSession(
+          this.selectedSession,
+          drawSessionStartingMarker
         );
-        sessionsUtils.onSingleSessionFetch(session, data, draw);
```

Comment:
I removed the nested callback from the callback and just call everything here. This method is kinda procedural, but at least everything is in one place and in order so it'll be easier to refactor this.